src/Controller/BackendCsvImportController.php

```diff
@@ -0,0 +1,243 @@
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    /**
+     * Generate the import view for list wizard
+     *
+     * @param DataContainer $dc
+     *
+     * @return string
+     *
+     * @throws RedirectResponseException
+     * @throws ResponseException
+     */
+    public function importListWizard(DataContainer $dc)
+    {
+        $csvImport = $this->createImportInstance();
+
+        $csvImport->setCallback(function (array $data, array $row) {
+            return array_merge($data, $row);
+        });
+
+        return $this->runDefaultRoutine($csvImport, $dc, 'listitems', $GLOBALS['TL_LANG']['MSC']['lw_import'][0]);
+    }
+
+    /**
+     * Generate the import view for option wizard
+     *
+     * @param DataContainer $dc
+     *
+     * @return string
+     *
+     * @throws RedirectResponseException
+     * @throws ResponseException
+     */
+    public function importOptionWizard(DataContainer $dc)
+    {
+        $csvImport = $this->createImportInstance();
+
+        $csvImport->setCallback(function (array $data, array $row) {
+            $data[] = [
+                'value' => $row[0],
+                'label' => $row[1],
+            ];
+
+            return $data;
+        });
+
+        return $this->runDefaultRoutine($csvImport, $dc, 'options', $GLOBALS['TL_LANG']['MSC']['ow_import'][0]);
+    }
+
+    /**
+     * Generate the import view for table wizard
+     *
+     * @param DataContainer $dc
+     *
+     * @return string
+     *
+     * @throws RedirectResponseException
+     * @throws ResponseException
+     */
+    public function importTableWizard(DataContainer $dc)
+    {
+        $csvImport = $this->createImportInstance();
+
+        $csvImport->setCallback(function (array $data, array $row) {
+            $data[] = $row;
+
+            return $data;
+        });
+
+        $csvImport->setSeparators([
+            CsvImportUtil::SEPARATOR_COMMA,
+            CsvImportUtil::SEPARATOR_SEMICOLON,
+            CsvImportUtil::SEPARATOR_TABULATOR,
+        ]);
+
+        return $this->runDefaultRoutine($csvImport, $dc, 'tableitems', $GLOBALS['TL_LANG']['MSC']['tw_import'][0]);
+    }
+
+    /**
+     * Create the CSV import instance
+     *
+     * @return CsvImportUtil
+     */
+    private function createImportInstance()
+    {
+        return new CsvImportUtil(
+            $this->connection,
+            $this->flashBag,
+            $this->requestStack->getCurrentRequest(),
+            $this->getFileUploader()
+        );
+    }
+
+    /**
+     * Run the default import routine
+     *
+     * @param CsvImportUtil $csvImport
+     * @param DataContainer $dc
+     * @param string        $fieldName
+     * @param string        $submitLabel
+     *
+     * @return string
+     *
+     * @throws RedirectResponseException
+     * @throws ResponseException
+     */
+    private function runDefaultRoutine(CsvImportUtil $csvImport, DataContainer $dc, $fieldName, $submitLabel)
```

Comment:
This should certainly not be `private`. Think of another developer trying to add an import wizard for their own widget, how will they be able to use this? I think it should rather look like this:

```php
class AbstractBackendCsvImportController
{
    protected function createImportInstance() {}
    protected function runDefaultRoutine(CsvImportUtil $csvImport, DataContainer $dc, $fieldName, $submitLabel) {}
}

TableWizardCsvImportBackendController extends AbstractBackendCsvImportController
{
    public function importTableWizard() {}
```